src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java

```diff
@@ -0,0 +1,457 @@
+
+import java.util.*;
+import java.util.Set;
+
+import org.geotools.data.FeatureSource;
+import org.geotools.data.Query;
+import org.geotools.factory.Hints;
+import org.geotools.feature.FeatureCollection;
+import org.geotools.feature.FeatureIterator;
+import org.geotools.geometry.jts.JTS;
+import org.geotools.referencing.ReferencingFactoryFinder;
+import org.joda.time.DateTime;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.referencing.crs.CRSAuthorityFactory;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opentripplanner.common.geometry.GeometryUtils;
+import org.opentripplanner.common.model.P2;
+import org.opentripplanner.graph_builder.impl.shapefile.AttributeFeatureConverter;
+import org.opentripplanner.graph_builder.impl.shapefile.StringAttributeFeatureConverter;
+import org.opentripplanner.graph_builder.services.GraphBuilder;
+import org.opentripplanner.graph_builder.services.shapefile.FeatureSourceFactory;
+import org.opentripplanner.graph_builder.services.shapefile.SimpleFeatureConverter;
+import org.opentripplanner.routing.alertpatch.Alert;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.TraverseMode;
+import org.opentripplanner.routing.core.TraverseModeSet;
+import org.opentripplanner.routing.edgetype.StreetEdge;
+import org.opentripplanner.routing.edgetype.StreetTraversalPermission;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Graph;
+import org.opentripplanner.routing.graph.Vertex;
+import org.opentripplanner.routing.impl.*;
+import org.opentripplanner.routing.services.GraphService;
+import org.opentripplanner.routing.services.PathService;
+import org.opentripplanner.routing.services.notes.StreetNotesService;
+import org.opentripplanner.routing.spt.GraphPath;
+import org.opentripplanner.routing.spt.ShortestPathTree;
+import org.opentripplanner.routing.vertextype.StreetVertex;
+import org.opentripplanner.routing.vertextype.IntersectionVertex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.MultiLineString;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+/**
+ * Loads an NIH shapefile into an edge-based graph, associating the edges with those previously
+ * imported from OpenStreetMap.
+ *
+ */
+public class NihShapefileStreetGraphBuilderImpl implements GraphBuilder {
+    private static Logger LOG = LoggerFactory.getLogger(ShapefileStreetGraphBuilderImpl.class);
+
+    private FeatureSourceFactory _featureSourceFactory;
+
+    private GraphServiceImpl graphService;
+
+    public List<String> provides() {
```

Comment:
Because it's providing an implementation, rather than subclassing, the way it works is that Java will throw an error if the function definition is missing from the implementation.